spec/faraday/request/multipart_spec.rb

```diff
@@ -68,6 +68,90 @@
     end
   end
 
+  context 'when providing json and IO content in the same payload' do
+    let(:io_content) { StringIO.new('io-content') }
+    let(:json_content) do
+      {
+        b: 1,
+        c: 2
+      }.to_json
+    end
+
+    let(:payload) do
+      {
+        json_content: Faraday::Request::PartParams.new(
+          json_content,
+          'Content-Type' => 'application/json'
+        ),
+        io_content: Faraday::UploadIO.new(io_content, 'application/pdf')
```

Comment:
A few inconsistencies stand out to me here.

1. Should the module namespace be `Faraday` or `Faraday::Request`? 
  * `Faraday::Request` is currently used by only the middleware classes themselves, like `Faraday::Request::Multipart`. It sure makes sense for `UploadIO` to live there too... I don't recall if I put it on `Faraday` because it'd be shorter for users to type, or it's part of the [multipart-post gem](https://github.com/socketry/multipart-post/blob/master/lib/composite_io.rb#L57-L108).
2. Hash vs content type argument. Since a `multipart-post` gem `ParamPart` [only uses `Content-Type` and `Content-ID`](https://github.com/socketry/multipart-post/blob/f119fc9739735ad4126e8edeaca1f292d039ba0c/lib/parts.rb#L52-L60), what do you think of a method signature like `PartParams.new(value, content_type, content_id=nil)`?

Reply:
I un-namespaced the `PartParams` class and used the method signature you suggested (it makes sense !!)